server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/CloudsBlobsConfiguration.java

```diff
@@ -0,0 +1,131 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.objectstorage;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.jclouds.openstack.keystone.config.KeystoneProperties;
+import org.jclouds.openstack.swift.v1.reference.TempAuthHeaders;
+import org.testcontainers.shaded.com.google.common.base.Preconditions;
+
+public class CloudsBlobsConfiguration {
+    public static class Builder {
+        private URI endpoint;
+        private Identity identity;
+        private Credentials credentials;
+        private Optional<Region> region;
+        private Optional<UserHeaderName> userHeaderName;
+        private Optional<PassHeaderName> passHeaderName;
+
+        public Builder() {
+            region = Optional.empty();
+            userHeaderName = Optional.empty();
+            passHeaderName = Optional.empty();
+        }
+
+        public CloudsBlobsConfiguration.Builder endpoint(URI endpoint) {
+            this.endpoint = endpoint;
+            return this;
+        }
+
+        public CloudsBlobsConfiguration.Builder identity(Identity identity) {
+            this.identity = identity;
+            return this;
+        }
+
+        public CloudsBlobsConfiguration.Builder credentials(Credentials credentials) {
+            this.credentials = credentials;
+            return this;
+        }
+
+        public CloudsBlobsConfiguration.Builder region(Region region) {
+            this.region = Optional.of(region);
+            return this;
+        }
+
+        public CloudsBlobsConfiguration.Builder tempAuthHeaderUserName(UserHeaderName tmpAuthHeaderUser) {
+            userHeaderName = Optional.of(tmpAuthHeaderUser);
+            return this;
+        }
+
+        public CloudsBlobsConfiguration.Builder tempAuthHeaderPassName(PassHeaderName tmpAuthHeaderPass) {
+            passHeaderName = Optional.of(tmpAuthHeaderPass);
+            return this;
+        }
+
+        public CloudsBlobsConfiguration build() {
+            Preconditions.checkState(endpoint != null);
+            Preconditions.checkState(identity != null);
+            Preconditions.checkState(credentials != null);
+            return new CloudsBlobsConfiguration(endpoint, identity, credentials, region, userHeaderName, passHeaderName);
+        }
+    }
+
+    private final URI endpoint;
+    private final Optional<Region> region;
+    private final Identity identity;
+    private final Credentials credentials;
+    private final Optional<UserHeaderName> userHeaderName;
+    private final Optional<PassHeaderName> passHeaderName;
+
+    CloudsBlobsConfiguration(URI endpoint,
```

Comment:
usually we add a visibleForTesting annotation to explicit the package-private visibility